src/features/filter.ts

```diff
@@ -1,12 +1,92 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Todo } from '../types/Todo';
 
-const initialState = {
+const initialState: FilterState = {
   query: '',
   status: 'all',
 };
 
+type Status = 'all' | 'active' | 'completed';
```

Comment:
Ensure that the `Status` type is used consistently across the application, especially in components that interact with the filter state, to maintain type safety.